huawei-wmi.c

```diff
@@ -391,7 +397,8 @@ static void huawei_wmi_leds_setup(struct device *dev)
 	huawei->cdev.dev = dev;
 	huawei->cdev.flags = LED_CORE_SUSPENDRESUME;
 
-	devm_led_classdev_register(dev, &huawei->cdev);
+	if (acpi_has_method(NULL, "\\SMLS") || (quirks && quirks->ec_micmute))
+		devm_led_classdev_register(dev, &huawei->cdev);
```

Comment:
Nice work checking `\SMLS`

This function `huawei_wmi_leds_setup` should handle all of the device LEDs micmute, Fn, etc. Perhaps change the `cdev` led name to `micmute_cdev`. Also I would cover the whole micmute led block in a if statement and implement a `huawei_wmi_leds_exit` function to unregister the leds.

```
huawei->micmute_led_available = false;
if (acpi_has_method("\\SMLS") || ...) {
  huawei->micmute_led_available = true;
  huawei->micmute_cdev.name = "platform::micmute";
  // ...
  // register led
}
// other less
```

Reply:
i would say for that, users should use the `ec_micmute` module param